src/ol/layer/Graticule.js

```diff
@@ -518,9 +532,15 @@ class Graticule extends VectorLayer {
    * resolution.
    * @param {import("../extent.js").Extent} extent Extent.
    * @param {number} resolution Resolution.
+   * @param {import("../proj/Projection.js").default} projection Projection.
+   * @param {import("../featureloader.js").FeatureLoaderInternal} [internal] Internal properties.
    * @return {Array<import("../extent.js").Extent>} Extents.
    */
-  strategyFunction(extent, resolution) {
+  strategyFunction(extent, resolution, projection, internal) {
```

Comment:
We have all this additional information in the frame state, so there should be no need to pass it to the strategy function.

Reply:
ol/source/Vector which calls the strategy function does not have the frameState, but at that point it does have the internal properties which will be passed to the loader.  Also the internal extent is not taken directly from the frameState as the renderer handles wrapping.